dev-libs/leveldb/files/leveldb-1.23-unbundle-gtest-benchmark.patch

```diff
@@ -0,0 +1,40 @@
+From: Arthur Zamarin <arthurzam@gmail.com>
+Date: Sat, 7 Aug 2021 20:29:45 +0300
+Subject: [PATCH] Unbundle GTest and benchmark
+
+use find_package to use the system installed GTest and benchmark
+
+Signed-off-by: Arthur Zamarin <arthurzam@gmail.com>
+
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -289,27 +289,8 @@ target_link_libraries(leveldbutil leveldb)
+ if(LEVELDB_BUILD_TESTS)
+   enable_testing()
+ 
+-  # Prevent overriding the parent project's compiler/linker settings on Windows.
```

Comment:
You can generally make the patch smaller by using `if (FALSE)` or alike. Or even better, by adding an `option()` to use system gtest/benchmark and making an upstreamable patch with it.

Reply:
Well, smart minds think a like, and found out that the idea was already thought about before, https://github.com/google/leveldb/pull/912.
I guess Google don't want this idea (and I can understand why actually, theirs reason is quite good).

Is it better a smaller patch (with the `if` clause) or use the one from the mentioned PR, whose patch is even bigger than mine (but better).

Reply:
Oh, right, it's gtest. In that case you don't even need to care to patch it. From my experience, using external gtest may cause real issues (e.g. because it heavily relies on matching flags).

Reply:
As obviously you are more experienced, I changed the way I use `gtest` and `benchmark`.
Now I don't unbundle them at all, and download the version specified as a submodule.
I decided to use the SRC_URI which downloads this commit, and then uses best matching `PV`, to have best chances to match other packages using the same commit and format.

Reply:
Argv, I am sorry, I didn't know that they aren't included in the tarball. I'd generally suggest doing whatever is easier, so up to you whether you prefer to fetch them or unbundle them.

Reply:
s/is easier/& and works/

Reply:
Then lets stay with current version (download `SRC_URI` and link) as it enables us to test like upstream, and like you said, no problems with mismatching flags. It also means less patch handling on Gentoo's side.